### DISCUSSION TITLE
Vendure vs other solution on PHP ?

### DISCUSSION THREAD

**Discussion by LeaKitty** [vendurehq/vendure#1668] [General]
Hi,

I have an e-commerce project to set up, and I have some questions :
- Has anyone ever done a solution comparison, for example Vendure vs Sylius (or other solution on PHP) to see the advantages and disadvantages of each ? 
Our teams include people developing in PHP and JS/TS

- Do you know a project if possible first "open source", and eventually a "paid" version to have a very advanced search bar for product search ?

Thanks.

**Comment by michaelbromley** [comment]
Hi,

* You can see a comparison between Vendure and several other solutions (including Sylius) here: https://headlesscommerceplatforms.com. My personal (and I guess biased) take on it is that a full-stack Node.js headless framework is superior to a PHP solution in a number of ways: you can use TypeScript which is IMO a better language to work with than PHP - its static typing is one of the most advanced of any language. Node.js apps tend to be easier to scale than PHP and less resource-intensive from my experience. And since you need to use JS/TS _anyway_ for the storefront part, it makes a lot of sense to keep the same skills, workflows & tools for the backend too.
* For Vendure we are planning to open up a marketplace for paid plugins later this year. One of the first paid plugins we will make available is a very advanced search solution, comparable to Algolia and based on the OSS search engine Typesense.
